ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -549,6 +549,67 @@ public func concatMap<T, U, E>(transform: T -> SignalProducer<U, E>)(producer: S
     return producer |> map(transform) |> concat
 }
 
+/// Returns a signal that forwards values from the latest signal sent on
+/// `producer`, ignoring values sent on previous inner signals.
+///
+/// An error sent on `producer` or the latest inner signal will be sent on the
+/// returned signal.
+///
+/// The returned signal completes when `producer` and the latest inner signal
+/// have both completed.
+public func latest<T, E>(producer: SignalProducer<SignalProducer<T, E>, E>) -> SignalProducer<T, E> {
+	return SignalProducer<T, E> { sink, disposable in
+		let producerCompleted = Atomic(false)
+		let latestCompleted = Atomic(false)
+		
+		let completeIfNecessary: () -> () = {
+			if producerCompleted.value && latestCompleted.value {
+				sendCompleted(sink)
+			}
+		}
+
+		let latestDisposable = SerialDisposable()
+		disposable.addDisposable(latestDisposable)
+
+		producer.startWithSignal { signal, producerDisposable in
+			disposable.addDisposable(producerDisposable)
+			
+			signal.observe(
+				next: { signal in
```

Comment:
Thanks for bringing attention to this, @neilpa. I was using [swift-before-signal-producers](https://github.com/ReactiveCocoa/ReactiveCocoa/releases/tag/swift-before-signal-producers) as a guide, and we use shadowing liberally there.

The argument I can imagine in favor of shadowing is that it ensures the shadowed referent isn't used inside the scope where it is shadowed, which makes the code easier to reason over in a way.

Re: referring to `SignalProducer`s as "signals", there's a lot of precedent all over the post-`ColdSignal` codebase, but I'm not sure if it's deliberate. 

I'd love to get more opinions from @ReactiveCocoa/reactivecocoa on both of these practices.